connectors/src/connectors/github/temporal/activities.ts

```diff
@@ -1206,6 +1292,16 @@ export async function githubCodeSyncActivity({
       logger.child({ task: "garbageCollectCodeSync" })
     );
 
+    // Create the Code folder node.
+    if (files.length > 0 || directories.length > 0) {
```

Comment:
We can create even if empty IMHO, since users might want to see the entry anyway (just like in your file explorer you see folders even if they are empty), wdyt?
Unless you did that for a specific reason?

Reply:
I did that to minimize the noise in the upcoming folder search, but not strongly opinionated, and it does add some logic